### PR TITLE
Validate the WEP name and assign from Spec fields if unassigned

### PR DESCRIPTION
**Pull request by robbrockbank**
Add some validation to the WEP name

### CODE REVIEW

File: lib/clientv2/workloadendpoint.go [hunk 1]
```diff
@@ -90,3 +99,34 @@ func (r workloadEndpoints) List(ctx context.Context, opts options.ListOptions) (
...
+		Endpoint: res.Spec.Endpoint,
+		Workload: res.Spec.Workload,
+		Pod: res.Spec.Pod,
+		ContainerID: res.Spec.ContainerID,
+	}
+	expectedName, err := wepids.CalculateWorkloadEndpointName(false)
+	if err != nil {
+		return err
+	}
+	if len(res.Name) == 0 {
```
Comment (gunjan5) [inline lib/clientv2/workloadendpoint.go]: curious why can't we do `res.Name == ""`? That seems more readable

Comment (gunjan5) [reply lib/clientv2/workloadendpoint.go]: i think it should have the same result but `len` is usually for a slice or array

Comment (robbrockbank) [reply lib/clientv2/workloadendpoint.go]: i think its what the k8s code does more often for empty string.
 should amount to the same thing

Comment (gunjan5) [reply lib/clientv2/workloadendpoint.go]: Ok, not too fussed about it. Just looks a bit odd

**Pull request merged**